### PR TITLE
Support fields "created" and "origin" for backup storages

### LINKED ISSUE #45: Support fields "created" and "origin" for backup storages
According to the API docs, backup storages will have additional information about when they were `created` and what their `origin` storage is:

* https://developers.upcloud.com/1.2/9-storages/#list-storages
* https://developers.upcloud.com/1.3/9-storages/#list-storages

> ```json
>       {
>         "access" : "private",
>         "created" : "2019-09-17T14:34:43Z",
>         "license" : 0,
>         "origin" : "01eff7ad-168e-413e-83b0-054f6a28fa23",
>         "size" : 10,
>         "state" : "online",
>         "title" : "On demand backup",
>         "type" : "backup",
>         "uuid" : "01287ad1-496c-4b5f-bb67-0fc2e3494740",
>         "zone" : "uk-lon1"
>       },
> ```

These fields seems to be missing in `upcloud-go-api`.

### PR DESCRIPTION
**Pull request by thomasheller**
Should fix #45. Let me know if any changes are needed.

### CODE REVIEW

File: upcloud/storage.go [hunk 1]
```diff
@@ -54,6 +57,9 @@ type Storage struct {
 	Type       string `xml:"type"`
 	UUID       string `xml:"uuid"`
 	Zone       string `xml:"zone"`
+	// Only for type "backup":
+	Origin  string    `xml:"origin"`
+	Created time.Time `xml:"created"`
```
Comment (Jalle19) [inline upcloud/storage.go]: The marshalling for these fields should probably be tested; IIRC there are tests like these that can be used as examples?

Comment (thomasheller) [reply upcloud/storage.go]: I just realized that the integration tests work on regular customer accounts, too, if `UPCLOUD_GO_SDK_TEST_USER` and `UPCLOUD_GO_SDK_TEST_PASSWORD` are provided. :tada: From the Travis error message I assumed that some special setup would be required.

Let me see if I can provide a useful test...

Comment (Jalle19) [reply upcloud/storage.go]: Be *very aware* that the integration test suite may delete all your resources if you're not careful, I suggest using a throwaway account. See https://github.com/UpCloudLtd/upcloud-go-api#testing.

Comment (thomasheller) [reply upcloud/storage.go]: Thanks for the pointer. I haven't set `UPCLOUD_GO_SDK_TEST_DELETE_RESOURCES` so `teardown()` won't be called. I'm also doing this on a non-production account. :slightly_smiling_face:

**Comment by thomasheller** [comment]
Added some tests. Let me know if anyhing needs to be changed.

File: upcloud/service/service_test.go [hunk 2]
```diff
@@ -407,14 +409,42 @@ func TestCreateBackup(t *testing.T) {
...
 
 	handleError(err)
 	waitForStorageOnline(storageDetails.UUID)
+
+	timeAfterBackup := utcTimeWithSecondPrecision()
+
 	t.Logf("Created backup with UUID %s", backupDetails.UUID)
+
+	// Get backup storage details
+	t.Logf("Getting details of backup sorage with UUID %s ...", backupDetails.UUID)
```
Comment (Jalle19) [inline upcloud/service/service_test.go]: Nitpick, typo in "storage"

**Comment by Jalle19** [comment]
@jgabor not sure who to ping, this repository usually doesn't get much love so maybe you could ping the right people?

File: upcloud/service/service_test.go (hunk 2 above)
Comment (thomasheller) [reply upcloud/service/service_test.go]: :+1: fixed

**Comment by jgabor** [comment]
> @jgabor not sure who to ping, this repository usually doesn't get much love so maybe you could ping the right people?

@Jalle19 Agree. I've pinged the right people.

**Comment by LordChocoBlizzBallz** [comment]
Thank you for the PR!

**Pull request merged**